weblate/memory/tasks.py

```diff
@@ -45,6 +46,72 @@ def import_memory(project_id: int, component_id: int | None = None) -> None:
                 handle_unit_translation_change(unit, None, component, project)
 
 
+@app.task(trail=False)
+def import_user_memory(user_id: int) -> None:
+    """
+    Import all translations from a user into their personal translation memory.
+
+    This is used when a user enables personal translation memory after having it disabled.
+    """
+    from weblate.auth.models import User
+    from weblate.trans.models import Unit
+
+    user = User.objects.get(pk=user_id)
+    if not user.profile.contribute_personal_tm:
+        return
+
+    with transaction.atomic():
+        units = (
+            Unit.objects.filter(
+                Q(change__author=user) | Q(change__user=user),
+                state__gte=STATE_TRANSLATED,
+                translation__component__is_glossary=False,
+            )
```

Comment:
This will return all units where the user has ever contributed, not just the ones where the translation he made is currently used. If we want this functionality, it would have to be done based on the `Change` model.